3.analyze_data/nbconverted/analyze_predicted_probabilities.py

```diff
@@ -0,0 +1,165 @@
+
+
+# ## Define and create the output paths
+
+# In[4]:
+
+
+filename = "comparison_results.parquet"
+output_path = pathlib.Path(f"{big_drive_path}/statistical_test_comparisons")
+output_path.mkdir(parents=True, exist_ok=True)
+
+
+# In[5]:
+
+
+treatment_paths["compound"]["platemap"]["broad_sample"].fillna("DMSO", inplace=True)
+
+
+# ## Mann-whitney U wrapper function
+
+# In[6]:
+
+
+def perform_mannwhitneyu_median(_dmso_probs, _treatment_probs):
+    """
+    Parameters
+    ----------
+    _dmso_probs: Pandas Series
+        The down-sampled predicted probilities of DMSO for a treatment type and phenotype.
+
+    _treatment_probs: Pandas Series
+        The predicted probabilities of the treatment.
+
+    Returns
+    -------
+    A zipped object which represents can be referenced by p_value and a comparison_metric_value, which are later on represented in the resulting dictionary.
+    """
+
+    test_result = mannwhitneyu(_dmso_probs, _treatment_probs, alternative="two-sided")
+    med_diff = _treatment_probs.median() - _dmso_probs.median()
+    return zip(["comparison_metric_value", "p_value"], [med_diff, test_result[1]])
+
+
+# ## Dunn wrapper function
+
+# In[7]:
+
+
+def perform_dunn_median(_dmso_probs, _treatment_probs):
+    """
+    Parameters
+    import numpy as np
+    ----------
+    _dmso_probs: Pandas Series
+        The down-sampled predicted probilities of DMSO for a treatment type and phenotype.
+
+    _treatment_probs: Pandas Series
+        The predicted probabilities of the treatment.
+
+    Returns
+    -------
+    A zipped object which represents can be referenced by p_value and a comparison_metric_value, which are later on represented in the resulting dictionary.
+    """
+    data = {
+        'probs': np.hstack((_dmso_probs.to_numpy(), _treatment_probs.to_numpy())),
+        'group': ['DMSO'] * len(_dmso_probs) + ['Treatment'] * len(_treatment_probs)
+    }
+
+    df = pd.DataFrame(data)
+    p_value = posthoc_dunn(df, val_col="probs", group_col="group")
+    med_diff = _treatment_probs.median() - _dmso_probs.median()
+    return zip(["comparison_metric_value", "p_value"], [med_diff, p_value.loc["DMSO", "Treatment"]])
+
+
+# ## Defining tests and aggregation metric names
+
+# In[8]:
+
+
+comp_functions = {"dunn_test":
+                  {"statistical_test_function": perform_dunn_median,
+                   "comparison_metric": "median_difference"},
+                  "mann_whitney_u":
+                  {"statistical_test_function": perform_mannwhitneyu_median,
+                   "comparison_metric": "median_difference"}}
+
+
+# In[9]:
+
+
+treatments = sig_test.get_treatment_comparison(comp_functions, treatment_paths, probadf)
+
+
+# ## Save the comparisons data
+
+# In[10]:
+
+
+treatments = pd.DataFrame(treatments)
+treatments.to_parquet(output_path / filename)
```

Comment:
In case you accepted the above suggestion for changing the file name:
```suggestion
treatments.to_parquet(output_path / comparison_results_output_filename)
```

Reply:
Changed!